lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemConfig.java

```diff
@@ -89,6 +89,7 @@ public static software.amazon.awssdk.core.retry.RetryMode getRetryMode(RetryMode
     private S3SseType sseType = S3SseType.NONE;
     private String sseKmsKeyId;
     private DataSize streamingPartSize = DataSize.of(16, MEGABYTE);
+    private int streamingMaxUploadThreads = Integer.MAX_VALUE;
```

Comment:
The native filesystem impl defaults to `Integer.MAX_VALUE` so I kept that the same. Ideas on better default?